src/Runner/Runner.php

```diff
@@ -109,20 +115,92 @@ public function __construct(
         $this->cacheManager = $cacheManager;
         $this->directory = $directory ?: new Directory('');
         $this->stopOnViolation = $stopOnViolation;
+        $this->parallel = $parallel;
     }
 
     public function fix(): array
     {
-        $changed = [];
+        $finderIterator = $this->finder instanceof \IteratorAggregate
+            ? $this->finder->getIterator()
+            : $this->finder;
 
-        $finder = $this->finder;
-        $finderIterator = $finder instanceof \IteratorAggregate ? $finder->getIterator() : $finder;
         $fileFilteredFileIterator = new FileFilterIterator(
             $finderIterator,
             $this->eventDispatcher,
             $this->cacheManager
         );
 
+        if ($this->parallel > 1) {
+            return $this->fixParallel($fileFilteredFileIterator);
+        }
+
+        return $this->fixProcedural($fileFilteredFileIterator);
+    }
+
+    private function fixParallel(FileFilterIterator $fileFilteredFileIterator): array
+    {
+        if (false === \function_exists('pcntl_fork')) {
+            return $this->fixProcedural($fileFilteredFileIterator);
+        }
+
+        $changed = [];
+        $childProcesses = [];
+
+        $files = iterator_to_array($fileFilteredFileIterator);
+        $numFiles = \count($files);
+        $numPerBatch = (int) ceil($numFiles / $this->parallel);
+
+        foreach (array_chunk($files, $numPerBatch) as $batch) {
+            $childOutFilename = tempnam(sys_get_temp_dir(), 'php-cs-fixer');
+            $pid = pcntl_fork();
+            if (-1 === $pid) {
+                exit('could not fork');
+            }
+            if ($pid) {
+                $childProcesses[$pid] = $childOutFilename;
+            } else {
+                $changedBatch = [];
+
+                /** @var \SplFileInfo $file */
+                foreach ($batch as $file) {
```

Comment:
this and content of `fixProcedural` is almost 1-1.
can be extracted to follow DRY